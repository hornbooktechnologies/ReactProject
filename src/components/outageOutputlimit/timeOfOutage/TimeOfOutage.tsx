import React, { useEffect, useState } from 'react'
import AnnualData from './annualData/AnnualData'
import PredictivePreviousData from './predictive&PreviousData/PredictivePreviousData'
import OutageHeader from '../OutageHeader'
import { useTranslation } from 'react-i18next'
import { GetAnnualTimeOutageValue, GetDonutChartValue, GetStartEndTimeOutageValue } from '../../../utils/outageOutputLimit/OutageOutputLimitUtils'
import { cssClassName } from '../../../utils/CssConstants'
import { AppConstant, OutageOutputConstant } from '../../../utils/AppConstants'
import { TimeOfOutageCardModel } from '../../../models/outageOutputLimit/TimeOfOutageModel'

const TimeOfOutage = (props :TimeOfOutageCardModel) => {
  const { i18n } = useTranslation();
  const [donutchartpresentToEndData, setDonutchartpresentToEndData] = useState<object | null >(null)
  const [donutchartstartToPresentData, setDonutchartstartToPresentData] = useState<object | null >(null)
  const [donutchartAnnualData, setDonutchartAnnualData] = useState<object | null >(null)
  const timeOutage = props?.timeOfOutageAtOutageData
  let annualData = timeOutage && timeOutage[0]
  let startTopresentData =  timeOutage && timeOutage[1]
  let presentToendData =  timeOutage && timeOutage[2]

  useEffect(() => {
    setDonutchartAnnualData(
      GetDonutChartValue(
        annualData?.plannedPlan,
        annualData?.plannedActualOrForcast,
        annualData?.plannedPreviousPlan,
        annualData?.plannedPreviousActualOrForcast,
        i18n.t("LABELS.HRS"),
        false,
        true
      )
    )
    setDonutchartpresentToEndData(
      GetDonutChartValue(
        presentToendData?.endPlannedPlan,
        presentToendData?.endPlannedActualOrForcast,
        presentToendData?.endPlannedPreviousPlan,
        presentToendData?.endPlannedPreviousActualOrForcast,
        i18n.t("LABELS.HRS"),
        false,
        true
      )
    )
    setDonutchartstartToPresentData(
      GetDonutChartValue(
        startTopresentData?.presentPlannedPlan,
        startTopresentData?.presentPlannedActualOrForcast,
        startTopresentData?.presentPlannedPreviousPlan,
        startTopresentData?.presentPlannedPreviousActualOrForcast,
        i18n.t("LABELS.HRS"),
        true,
        true
      )
    )
  }, [presentToendData,startTopresentData,annualData,i18n])

  const plannedPositiveList = [
    GetStartEndTimeOutageValue(
      cssClassName.MB_20,
      cssClassName.JERA_DISLIKE_ICON_CLASS,
      OutageOutputConstant.INCREASED_PLANED,
      presentToendData?.endPlannedIncreseHrs,
      presentToendData?.endPlannedPreviousIncreseHrs,
      presentToendData?.endPlannedIncreseRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.POSITIVE_VAL),
  
      GetStartEndTimeOutageValue(
       '',
       cssClassName.JERA_DISLIKE_ICON_CLASS,
       OutageOutputConstant.UNPLANNED_OUTAGE,
       presentToendData?.endUnplannedHrs,
       presentToendData?.endUnplannedPreviousHrs,
       presentToendData?.endUnplannedRecords,
       i18n.t("LABELS.HRS"),
       OutageOutputConstant.POSITIVE_VAL
      ),
  ]
  const plannedNegativeList = [
    GetStartEndTimeOutageValue(
      cssClassName.MB_20,
      cssClassName.JERA_LIKE_ICON_CLASS,
      OutageOutputConstant.REDUCED_PLANED,
      presentToendData?.endPlannedDecreseHrs,
      presentToendData?.endPlannedPreviousDecreseHrs,
      presentToendData?.endPlannedDecreseRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.NEGATIVE_VAL
    ),
    GetStartEndTimeOutageValue(
      '',
      cssClassName.JERA_LIKE_ICON_CLASS,
      OutageOutputConstant.CANCELLED_PLANED,
      presentToendData?.endCancledHrs,
      presentToendData?.endCancledPreviousHrs,
      presentToendData?.endCancledRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.NEGATIVE_VAL
    ),
  ]
  const plannedPositiveStartTopresentList = [
    GetStartEndTimeOutageValue(
      cssClassName.MB_20,
      cssClassName.JERA_DISLIKE_ICON_CLASS,
      OutageOutputConstant.INCREASED_PLANED,
      startTopresentData?.presentPlannedIncreseHrs,
      startTopresentData?.presentPlannedPreviousIncreseHrs,
      startTopresentData?.presentPlannedIncreseRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.POSITIVE_VAL
    ),
    GetStartEndTimeOutageValue(
      '',
      cssClassName.JERA_DISLIKE_ICON_CLASS,
      OutageOutputConstant.UNPLANNED_OUTAGE,
      startTopresentData?.presentUnplannedHrs,
      startTopresentData?.presentUnplannedPreviousHrs,
      startTopresentData?.presentUnplannedRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.POSITIVE_VAL
    ),
  ]
  const plannedNegativeStartTopresentList = [
    GetStartEndTimeOutageValue(
      cssClassName.MB_20,
      cssClassName.JERA_LIKE_ICON_CLASS,
      OutageOutputConstant.REDUCED_PLANED,
      startTopresentData?.presentPlannedDecreseHrs,
      startTopresentData?.presentPlannedPreviousDecreseHrs,
      startTopresentData?.presentPlannedDecreseRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.NEGATIVE_VAL
    ),
    GetStartEndTimeOutageValue(
      '',
      cssClassName.JERA_LIKE_ICON_CLASS,
      OutageOutputConstant.CANCELLED_PLANED,
      startTopresentData?.presentCancledHrs,
      startTopresentData?.presentCancledPreviousHrs,
      startTopresentData?.presentCancledRecords,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.NEGATIVE_VAL
    )
  ]
  const annualImpact = [
    GetAnnualTimeOutageValue(
      annualData?.positvePreviousImpactHrs,
      annualData?.positveImpactHrs,
      i18n.t("LABELS.HRS"),
      OutageOutputConstant.POSITIVE_VAL
    ),
    GetAnnualTimeOutageValue(
      annualData?.NagetivePreviousImpactHrs,
      annualData?.NagetiveImpactHrs,
      i18n.t("LABELS.HRS"),
      AppConstant.MINUS_SYMBOL
    ),
  ]
  return (
    <>
      <OutageHeader
          title={i18n.t('OUTAGE_OUTPUTLIMIT_TITLE.TIME_OF_OUTAGE')}
        />
        <div className="uk-grid ml--25">
        <AnnualData
           donutchartAnnualData={donutchartAnnualData}
           annualImpact={annualImpact}
          />
         <PredictivePreviousData 
            donutchartData4_1={donutchartstartToPresentData}
            donutchartData3_31={donutchartpresentToEndData}
            plannedPositive={plannedPositiveList}
            plannedNegative={plannedNegativeList}
            plannedPositiveStartTopresent={plannedPositiveStartTopresentList}
            plannedNegativeStartTopresent={plannedNegativeStartTopresentList}
          />
      </div>
    </>
  )
}

export default TimeOfOutage;
