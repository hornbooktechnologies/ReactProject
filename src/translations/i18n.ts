import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
