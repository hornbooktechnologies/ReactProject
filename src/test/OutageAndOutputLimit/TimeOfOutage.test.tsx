import React from "react";
import renderer from "react-test-renderer";
import TimeOfOutage from "../../components/outageOutputlimit/timeOfOutage/TimeOfOutage";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: TimeOfOutage", () => {
  it("TimeOfOutage renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <TimeOfOutage/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
