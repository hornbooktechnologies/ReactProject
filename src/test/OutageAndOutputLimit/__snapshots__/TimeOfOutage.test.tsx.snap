// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TimeOfOutage TimeOfOutage renders correctly 1`] = `
Array [
  <h5
    className="mb-24 font-18"
  >
    OUTAGE_OUTPUTLIMIT_TITLE.TIME_OF_OUTAGE
  </h5>,
  <div
    className="uk-grid ml--25"
  >
    <div
      className="uk-width-1-3@m media-mb-20"
    >
      <div
        className="uk-card uk-height-1-1"
      >
        <h5
          className="mb-24 font-18"
        >
          LABELS.ANNUAL
        </h5>
        <div
          className="mb-40 ml--25 uk-grid uk-child-width-1-2@s uk-child-width-1-2@m"
          uk-grid="true"
        >
          <div
            className="media-mb-20"
          >
            <div
              className="uk-flex uk-flex-middle"
            >
              <span
                className="uk-up-icon uk-text-center uk-inline mr-10"
              >
                <i
                  className="jera uk-position-center font-32 color-green-60 jera-like-1"
                />
              </span>
              <h6
                className="uk-margin-remove font-14 color-green-60"
              >
                +
                OUTAGE_AND_OUTPUT_LIMIT.IMPACT
                <div>
                  <span
                    className="mt-5 mb-0 uk-text-bold font-14 
                      color-white
                       "
                  >
                    +
                    <span
                      className="font-8 uk-text-normal uk-text-capitalize"
                    >
                      LABELS.HRS
                    </span>
                  </span>
                </div>
              </h6>
            </div>
          </div>
          <div
            className="media-mb-20"
          >
            <div
              className="uk-flex uk-flex-middle"
            >
              <span
                className="uk-up-icon uk-text-center uk-inline mr-10"
              >
                <i
                  className="jera uk-position-center font-32 color-red-60 jera-dislike"
                />
              </span>
              <h6
                className="uk-margin-remove font-14 color-red-60"
              >
                -
                OUTAGE_AND_OUTPUT_LIMIT.IMPACT
                <div>
                  <span
                    className="mt-5 mb-0 uk-text-bold font-14 
                      color-white
                       "
                  >
                    -
                    <span
                      className="font-8 uk-text-normal uk-text-capitalize"
                    >
                      LABELS.HRS
                    </span>
                  </span>
                </div>
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="uk-width-expand@m"
    >
      <div
        className="uk-card mb-20"
      >
        <h5
          className="mb-24 font-18"
        >
          LABELS.YEAR_START_TO_PRESENT
        </h5>
        <div
          className="uk-grid uk-grid-small uk-child-width-1-3@s uk-child-width-1-3@m mm-10"
          uk-grid="true"
        >
          <div
            className="media-mb-20 pl-10"
          >
            <div>
              <div
                className="uk-flex uk-flex-middle mb-20"
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-green-60 jera-like-1 uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Reduced Planned Outage Time
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                className="uk-flex uk-flex-middle "
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-green-60 jera-like-1 uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Cancelled Planned Outage
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pl-10"
          >
            <div>
              <div
                className="uk-flex uk-flex-middle mb-20"
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-red-60 jera-dislike uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Increased Planned Outage Time
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      +
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                className="uk-flex uk-flex-middle "
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-red-60 jera-dislike uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Unplanned Outage
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      +
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="uk-text-center mt--35"
          >
            <div
              className="uk-flex uk-flex-center uk-flex-middle uk-flex-column chart-100 uk-margin-auto-left uk-margin-auto-right"
            />
          </div>
        </div>
      </div>
      <div
        className="uk-card"
      >
        <h5
          className="mb-24 font-18"
        >
          LABELS.PRESENT_TO_YEAR_END
        </h5>
        <div
          className="uk-grid uk-grid-small uk-child-width-1-3@s uk-child-width-1-3@m mm-10"
          uk-grid="true"
        >
          <div
            className="media-mb-20 pl-10"
          >
            <div>
              <div
                className="uk-flex uk-flex-middle mb-20"
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-green-60 jera-like-1 uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Reduced Planned Outage Time
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                className="uk-flex uk-flex-middle "
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-green-60 jera-like-1 uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Cancelled Planned Outage
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pl-10"
          >
            <div>
              <div
                className="uk-flex uk-flex-middle mb-20"
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-red-60 jera-dislike uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Increased Planned Outage Time
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      +
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                className="uk-flex uk-flex-middle "
              >
                <span
                  className="uk-up-icon uk-text-center uk-inline mr-5"
                >
                  <i
                    className="jera icn-24 color-red-60 jera-dislike uk-position-center"
                  />
                </span>
                <div>
                  <p
                    className="uk-margin-remove font-14"
                  >
                    Unplanned Outage
                  </p>
                  <span
                    className="mb-0 uk-text-bold font-14 mt-5"
                  >
                    <span
                      className=""
                    >
                      +
                      <span
                        className="font-8 uk-text-normal ml-3"
                      >
                        LABELS.HRS
                      </span>
                    </span>
                    <span
                      className="font-8 uk-text-normal ml-3"
                    >
                      <span
                        className="uk-text-bold font-12"
                      >
                        (
                        <span
                          className="font-8"
                        >
                          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
                        </span>
                        )
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="uk-text-center mt--35"
          >
            <div
              className="uk-flex uk-flex-center uk-flex-middle uk-flex-column chart-100 uk-margin-auto-left uk-margin-auto-right"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
